contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1469,7 +1496,7 @@ pltsql_strpos_non_determinstic(text *src_text, text *substr_text, Oid collid, in
 							u_errorName(status))));
 
 			/* for CS_AI collations usearch can give false positives so we double check the results here */
-			if (!(is_CS_AI && icu_compare_utf8_coll(mylocale, &src_uchar[usearch_getMatchedStart(usearch)], usearch_getMatchedLength(usearch), substr_uchar, substr_ulen) != 0))
+			if (!(is_CS_AI && icu_compare_utf8_coll(mylocale->info.icu.ucol, &src_uchar[usearch_getMatchedStart(usearch)], usearch_getMatchedLength(usearch), substr_uchar, substr_ulen, false) != 0))
```

Comment:
save extra call of usearch_getMatchedStart

Reply:
It is actually different calls
```&src_uchar[usearch_getMatchedStart(usearch)], usearch_getMatchedLength(usearch)```